test/functional/feature_startupnotify.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test -startupnotify."""
+
+import os
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+NODE_DIR = "node0"
+FILE_NAME = "test.txt"
+
+
+class StartupNotifyTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.disable_syscall_sandbox = True
+
+    def run_test(self):
+        tmpdir = os.path.join(self.options.tmpdir, NODE_DIR, FILE_NAME)
```

Comment:
Why the variable is called `tmpdir` if it's actually filename?

Reply:
this is the temporary directory where the file is...